js/classes/entities/midground/weapon/projectile_factory.js

```diff
@@ -0,0 +1,56 @@
+class ProjectileFactory {
+    static create(type, startPos, targetPos) {
+        switch (type) {
+            case ProjectileFactory.ROCK:
+                return new Rock(startPos, targetPos);
+            case ProjectileFactory.BOMB:
+                return new Bomb(startPos, targetPos);
+            case ProjectileFactory.SNOWBALL:
+                return new Snowball(startPos, targetPos);
+            case ProjectileFactory.SLIMEBALL:
+                return new Slimeball(startPos, targetPos);
+            case ProjectileFactory.LASER:
+                return new Laser(startPos, targetPos);
+            case ProjectileFactory.SONIC_WAVE:
+                return new SonicWave(startPos, targetPos);
+            case ProjectileFactory.BROCCOLI:
+                return new Broccoli(startPos, targetPos);
+            case ProjectileFactory.SUS_SNOWBALL:
+                return new SusSnowball(startPos, targetPos);
+        }
```

Comment:
I think a switch is overkill here, perhaps something like this would work?

```js
const projectileMap = {
    [ProjectileFactory.ROCK]: Rock,
    [ProjectileFactory.BOMB]: Bomb,
    [ProjectileFactory.SNOWBALL]: Snowball,
    [ProjectileFactory.SLIMEBALL]: Slimeball,
    [ProjectileFactory.LASER]: Laser,
    [ProjectileFactory.SONIC_WAVE]: SonicWave,
    [ProjectileFactory.BROCCOLI]: Broccoli,
    [ProjectileFactory.SUS_SNOWBALL]: SusSnowball,
};

return new (projectileMap[type])(startPos, targetPos);
```

Reply:
Agreed.

Reply:
Great idea! Thank you